src/OpenTelemetry.AutoInstrumentation.StartupHook/RulesEngine/RuleEngine.cs

```diff
@@ -28,23 +28,8 @@ internal class RuleEngine
         new MinSupportedFrameworkRule()
     };
 
-    private readonly List<Rule> _otherRules = new()
-    {
-        new OpenTelemetrySdkMinimumVersionRule(),
-        new DiagnosticSourceRule(),
-        new InstrumentationAssemblyRule(),
-        new NativeProfilerDiagnosticsRule()
-    };
-
-    internal RuleEngine()
-    {
-    }
-
-    // This constructor is used for test purpose.
-    internal RuleEngine(List<Rule> rules)
-    {
-        _otherRules = rules;
-    }
```

Comment:
Did you need to change the way the rules are injected? You probably could have kept the existing injection pattern but added a new private field for the `_optionalTests` and made `_otherRules` static. Then you could update the .ctors to do something like the following.
```
internal RuleEngine() : this(_otherRules) {}

//This constructor is used for test purpose
internal RuleEngine(List<Rule> optionalRules)
{
    _optionalRules = optionalRules;
}
```

Reply:
It is not required to change the way that the injection happens as long as the list of optional rules is only created when it is going to be used. Let me play with it.

Reply:
Returned to prior injection pattern.